### DISCUSSION TITLE
Is dataset and queryset always same on same scale unit?

### DISCUSSION THREAD

**Discussion by cow12331** [TPC-Council/HammerDB#529] [General]
I can't find how the data set and query set generate. Is the data set and query set always the same on different databases?

**Comment by sm-shaw** [comment]
The question is not entirely clear, but the assumption is that you are asking about TPROC-H? To see how the schema is built, run the build and browse the script in the script editor. Similarly, you can do the same when running the test. Note these both load the tpchcommon-1.0 module in the modules directory that contains functions common to both build and test for all databases. 

Therefore, the implementation is the same but for the build the dataset is not going to be identical even for 2 builds on the same database as there is a random element to the data generated. Similarly, for the query set, the order of queries can be seen in the module (this is the order given in the TPC-H specification). The queries themselves have some of the predicates completed at runtime, so this is a random element, meaning again even on the same database no 2 runs are identical. 

 So it should be clear that a lot of the code is shared, but no schema build or test will run the same thing exactly twice by design. (Otherwise, you could cache the result). And of course, the same queries may (and will) behave very differently on different databases due to how the queries are processed. 

And it is all open source and in clear text so you can see exactly what is being run by looking in the script editor in the GUI or print script in the CLI.
